### PR TITLE
feat(obstacle_cruise_planner): use motion utils

### PR DESCRIPTION
**Pull request by purewater0901**
Signed-off-by: yutaka <purewater0901@gmail.com>

## Description

Use insert stop point from motion utils to get the accurate stop distance

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/obstacle_cruise_planner/src/planner_interface.cpp [hunk 1]
```diff
@@ -130,37 +130,33 @@ Trajectory PlannerInterface::generateStopTrajectory(
...
   // Generate Output Trajectory
   auto output_traj = planner_data.traj;
-  for (size_t o_idx = zero_vel_idx; o_idx < output_traj.points.size(); ++o_idx) {
-    output_traj.points.at(o_idx).longitudinal_velocity_mps = 0.0;
+  const double zero_vel_dist =
+    closest_obstacle_dist - vehicle_info_.max_longitudinal_offset_m - feasible_margin_from_obstacle;
+  const auto zero_vel_idx = motion_utils::insertStopPoint(0, zero_vel_dist, output_traj.points);
+  if (zero_vel_idx) {
+    const auto wall_idx = obstacle_cruise_utils::getIndexWithLongitudinalOffset(
+      planner_data.traj.points, vehicle_info_.max_longitudinal_offset_m, *zero_vel_idx);
```
Comment (takayuki5168) [inline planning/obstacle_cruise_planner/src/planner_interface.cpp:140]: use abs_ego_offset

File: planning/obstacle_cruise_planner/src/planner_interface.cpp [hunk 2]
```diff
@@ -130,37 +130,33 @@ Trajectory PlannerInterface::generateStopTrajectory(
...
-  const size_t wall_idx = obstacle_cruise_utils::getIndexWithLongitudinalOffset(
-    planner_data.traj.points, -feasible_margin_from_obstacle, collision_idx);
-
-  // TODO(shimizu) insert stop point with interpolation
   // Generate Output Trajectory
   auto output_traj = planner_data.traj;
-  for (size_t o_idx = zero_vel_idx; o_idx < output_traj.points.size(); ++o_idx) {
-    output_traj.points.at(o_idx).longitudinal_velocity_mps = 0.0;
+  const double zero_vel_dist =
+    closest_obstacle_dist - vehicle_info_.max_longitudinal_offset_m - feasible_margin_from_obstacle;
```
Comment (takayuki5168) [inline planning/obstacle_cruise_planner/src/planner_interface.cpp:136]: use abs_ego_offset

File: planning/obstacle_cruise_planner/src/planner_interface.cpp [hunk 3]
```diff
@@ -130,37 +130,33 @@ Trajectory PlannerInterface::generateStopTrajectory(
...
+
+    // Publish Stop Reason
+    const auto stop_pose = output_traj.points.at(*zero_vel_idx).pose;
+    const auto stop_reasons_msg =
+      makeStopReasonArray(planner_data.current_time, stop_pose, *closest_stop_obstacle);
+    stop_reasons_pub_->publish(stop_reasons_msg);
+
+    // Publish if ego vehicle collides with the obstacle with a limit acceleration
+    const auto stop_speed_exceeded_msg =
+      createStopSpeedExceededMsg(planner_data.current_time, will_collide_with_obstacle);
```
Comment (takayuki5168) [inline planning/obstacle_cruise_planner/src/planner_interface.cpp:157]: use output_traj.points for wall_idx handling.

File: planning/obstacle_cruise_planner/src/planner_interface.cpp (hunk 1 above)
Comment (purewater0901) [reply planning/obstacle_cruise_planner/src/planner_interface.cpp]: done

File: planning/obstacle_cruise_planner/src/planner_interface.cpp (hunk 2 above)
Comment (purewater0901) [reply planning/obstacle_cruise_planner/src/planner_interface.cpp]: done

File: planning/obstacle_cruise_planner/src/planner_interface.cpp (hunk 3 above)
Comment (purewater0901) [reply planning/obstacle_cruise_planner/src/planner_interface.cpp]: done

**Comment by purewater0901** [comment]
@takayuki5168 Please merge this PR after [this fix](https://github.com/autowarefoundation/autoware.universe/pull/1397)

**Review by takayuki5168 — APPROVED** [review]
LGTM

**Comment by takayuki5168** [comment]
@purewater0901 Resolve the conflicts please.

**Comment by purewater0901** [comment]
fixed

**Pull request merged**